From 064b4b8cf237890b87dc81565296b60f76f799f0 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 20 Nov 2024 19:42:57 -0600
Subject: [PATCH 1/4] Fix Wayland appId

Ensure the .desktop file matches the application-id, this fixes
window/icon associations on Wayland. Without it the texworks icon
may not be associated with the texworks open windows for example.
---
 res/{texworks.appdata.xml => org.tug.texworks.appdata.xml} | 0
 res/{texworks.desktop => org.tug.texworks.desktop}         | 0
 src/CMakeLists.txt                                         | 4 ++--
 3 files changed, 2 insertions(+), 2 deletions(-)
 rename res/{texworks.appdata.xml => org.tug.texworks.appdata.xml} (100%)
 rename res/{texworks.desktop => org.tug.texworks.desktop} (100%)

diff --git a/res/texworks.appdata.xml b/res/org.tug.texworks.appdata.xml
similarity index 100%
rename from res/texworks.appdata.xml
rename to res/org.tug.texworks.appdata.xml
diff --git a/res/texworks.desktop b/res/org.tug.texworks.desktop
similarity index 100%
rename from res/texworks.desktop
rename to res/org.tug.texworks.desktop
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5538d110..391dbf1a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -330,8 +330,8 @@ IF ( UNIX AND NOT APPLE )
     ${TeXworks_SOURCE_DIR}/NEWS
   DESTINATION ${CMAKE_INSTALL_PREFIX}/${TeXworks_DOCS_DIR})
   INSTALL(FILES ${TeXworks_SOURCE_DIR}/res/texworks.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/${TeXworks_MAN_DIR})
-  INSTALL(FILES ${TeXworks_SOURCE_DIR}/res/texworks.desktop DESTINATION ${CMAKE_INSTALL_PREFIX}/${TeXworks_DESKTOP_DIR})
-  INSTALL(FILES ${TeXworks_SOURCE_DIR}/res/texworks.appdata.xml DESTINATION ${CMAKE_INSTALL_PREFIX}/${TeXworks_APPDATA_DIR})
+  INSTALL(FILES ${TeXworks_SOURCE_DIR}/res/org.tug.texworks.desktop DESTINATION ${CMAKE_INSTALL_PREFIX}/${TeXworks_DESKTOP_DIR})
+  INSTALL(FILES ${TeXworks_SOURCE_DIR}/res/org.tug.texworks.appdata.xml DESTINATION ${CMAKE_INSTALL_PREFIX}/${TeXworks_APPDATA_DIR})
 ENDIF ( UNIX AND NOT APPLE )
 
 # On OS X, ensure there is always a `qt.conf` file as all hell will break loose
-- 
2.47.0

